recipes/tree-sitter-sql/all/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 3.15)
+project(tree-sitter-sql LANGUAGES C)
+
+find_package(tree-sitter REQUIRED CONFIG)
+include(GenerateExportHeader)
+
+file(WRITE api.h [[
```

Comment:
I don't think it's needed to create this file, the project offers a very similar one:

`./bindings/c/tree-sitter-sql.h:const TSLanguage *tree_sitter_sql(void);`

Reply:
@uilianries 
Thank you for your review.
I know your pointed file.

The main difference is api.h includes generated `tree_sitter_sql_export.h` as following:
```c
#ifndef TREE_SITTER_C_EXPORT_H
#define TREE_SITTER_C_EXPORT_H

#ifdef TREE_SITTER_C_STATIC_DEFINE
#  define TREE_SITTER_C_EXPORT
#  define TREE_SITTER_C_NO_EXPORT
#else
#  ifndef TREE_SITTER_C_EXPORT
#    ifdef tree_sitter_sql_EXPORTS
        /* We are building this library */
#      define TREE_SITTER_C_EXPORT 
#    else
        /* We are using this library */
#      define TREE_SITTER_C_EXPORT 
#    endif
#  endif

#  ifndef TREE_SITTER_C_NO_EXPORT
#    define TREE_SITTER_C_NO_EXPORT 
#  endif
#endif

#ifndef TREE_SITTER_C_DEPRECATED
#  define TREE_SITTER_C_DEPRECATED __attribute__ ((__deprecated__))
#endif

#ifndef TREE_SITTER_C_DEPRECATED_EXPORT
#  define TREE_SITTER_C_DEPRECATED_EXPORT TREE_SITTER_C_EXPORT TREE_SITTER_C_DEPRECATED
#endif

#ifndef TREE_SITTER_C_DEPRECATED_NO_EXPORT
#  define TREE_SITTER_C_DEPRECATED_NO_EXPORT TREE_SITTER_C_NO_EXPORT TREE_SITTER_C_DEPRECATED
#endif

/* NOLINTNEXTLINE(readability-avoid-unconditional-preprocessor-if) */
#if 0 /* DEFINE_NO_DEPRECATED */
#  ifndef TREE_SITTER_C_NO_DEPRECATED
#    define TREE_SITTER_C_NO_DEPRECATED
#  endif
#endif

#endif /* TREE_SITTER_C_EXPORT_H */
```

It helps to build windows shared build.

As I wrote the thread of this PR, I just followed tree-sitter-c style.
I want CCI's community to define the standard style of tree-sitter's syntax recipe.

If it is defined, I'm going to create several recipes(C++, CQL, CMake,...) about tree-sitter following that standard style later.

Reply:
@toge Sorry, I didn't read completely that header file. Indeed you did a great job investigating that project :smile: 

I would not recommend creating that custom header, even if it's case of sacrificing shared library support on Windows, because it will result not only in an unique new header, but we will become the maintainers. The project itself is a bit limited in terms of building, as you noted, so don't worry trying to add extra support.